visidata/loaders/_pandas.py

```diff
@@ -17,6 +17,9 @@ def __getitem__(self, k):
         return self.df.iloc[k]
 
     def __getattr__(self, k):
+        # This check helps avoid infinite recursion trouble
+        if k == 'df':
+            raise AttributeError
```

Comment:
```
if hasattr(y, '__setstate__'):
```

I bumped into, too!

This occurs because we do not handle missing attributes. I do not think the solution is necessarily to hardcode `if k == 'df'`. Instead, we should check if self has the attribute `df`, and raise AttributeError otherwise.

Like here:
https://github.com/saulpw/visidata/pull/491/files

Reply:
@saulpw, we should audit the code for all implementations of `__getattr__`, and ensure that we handle what happens when attributes are missing.

Reply:
True @anjakefala - calling out `df` explicitly looks hacky and I'm sure we can make it cleaner. It would probably make sense to have `DataFrameAdapter` be a `pd.DataFrame` subclass if not for the initial loading time. The way you linked could work and be more flexible (in case we need to add other local attributes someday).